modules/apps/oauth2-provider/oauth2-provider-rest/src/main/java/com/liferay/oauth2/provider/rest/internal/spi/bearer/token/provider/DefaultBearerTokenProvider.java

```diff
@@ -56,6 +56,11 @@ public void onBeforeCreate(AccessToken accessToken) {
 			generateTokenKey(
 				_defaultBearerTokenProviderConfiguration.
 					accessTokenKeyByteSize()));
+
+		if (_defaultBearerTokenProviderConfiguration.issueJWTAccessToken()) {
+			accessToken.setTokenType(
+				BearerTokenProviderConstants.TOKEN_TYPE_JWT_AT);
```

Comment:
THis is really weird design, you have to partially influence one of BearerTokenProvider(in this case, defaultBearerTokenPRovider), and expect latter code to execute based on the result.

Reply:
If it is weird or not I think depends largely on two factors: 
* How obvious is it when looking at the SPI that setTokenType() influences the token generation. p.s. I initially used an enum for this as it would enforce the discreet values, but that was a major SemVer change (maybe it should be, actually)
* If you expect DXP to be a platform or product

Note that is the process flow we have in SAML. See https://github.com/liferay-appsec/liferay-portal/pull/670/commits/e1f8e6c20c6032af9c4b796b7f98dfca707d6ed1?short_path=d6b385e#diff-d6b385ef9c3b48dc94a6742faff5432a6d7c29b9cfae310cebfe18900ccf8e24